packages/ember/src/components/form/select.gts

```diff
@@ -0,0 +1,445 @@
+// https://github.com/glimmerjs/glimmer.js/issues/408
+import { tracked, cached } from '@glimmer/tracking';
+// @ts-expect-error Ember keyboard doesn't currently ship a type for the `on-key` modifier
+// https://github.com/adopted-ember-addons/ember-keyboard/issues/464
+import onKey from 'ember-keyboard/modifiers/on-key';
+import { runTask } from 'ember-lifeline';
+
+import BoundValue from './bound-value.ts';
+import onInsert from '../../modifiers/did-insert.ts';
+
+import type { Optional } from '../../types.d.ts';
+import type IntlService from 'ember-intl/services/intl';
+
+declare type SelectOption<T> = {
+  label: string;
+  value: string | T;
+  raw: T;
+};
+
+interface SelectItemSignature<T> {
+  Args: {
+    optionIndex: number;
+    activeIndex: number;
+    currentValue: string | T;
+    option: SelectOption<T>;
+  };
+  Blocks: {
+    default: [];
+  };
+  Element: HTMLLIElement;
+}
+
+class SelectItem<T> extends Component<SelectItemSignature<T>> {
+  get classList() {
+    const classes = ['dropdown-item'];
+
+    const useIndexActive = this.args.activeIndex != -1;
+    const isCurrentIndex = isEqual(
+      this.args.optionIndex,
+      this.args.activeIndex,
+    );
+    const isCurrentValue = isEqual(
+      this.args.option.value,
+      this.args.currentValue,
+    );
+
+    if (useIndexActive && isCurrentIndex) {
+      classes.push('active');
+    }
+
+    if (!useIndexActive && isCurrentValue) {
+      classes.push('active');
+    }
+
+    return classes.join(' ');
+  }
+
+  get isActive() {
+    return isEqual(this.args.option.value, this.args.currentValue);
+  }
+
+  <template>
+    <li
+      class={{this.classList}}
+      role="option"
+      aria-selected={{this.isActive}}
+      ...attributes
+    >
+      {{yield}}
+    </li>
+  </template>
+}
+
+export interface SelectSignature<T> {
+  Args: {
+    disabled?: boolean;
+    loading?: boolean;
+    scrollable?: boolean;
+    defaultText?: string;
+    options: T[];
+    displayPath?: string;
+    serializationPath?: string | null;
+  };
+  Blocks: {
+    display?: [T | undefined];
+    option?: [T | undefined];
+    empty?: [];
+  };
+  Element: HTMLButtonElement;
+}
+
+export default class Select<T> extends BoundValue<
+  SelectSignature<T>,
+  string | T
+> {
+  @tracked
+  _isOpen = false;
+
+  @tracked
+  menuId = crypto.randomUUID();
+
+  @tracked
+  menuElement: Optional<HTMLElement> = null;
+
+  @tracked
+  activeItem = -1;
+
+  @tracked
+  internalSearchBuffer = '';
+
+  @service
+  declare intl: IntlService;
+
+  get classList() {
+    const classes = ['dropdown', 'form-control', 'text-start', 'focus-ring'];
+
+    if (this.scrollable) {
+      classes.push('scrollable');
+    }
+
+    return classes.join(' ');
+  }
+
+  get caretIcon() {
+    return this.isOpen ? 'bi-caret-up' : 'bi-caret-down-fill';
+  }
+
+  get defaultText() {
+    const baseDefaultText = this.intl.t('nrg.select.baseDefaultText', {});
```

Comment:
Could this not just be `defaultText`, like the argument? It's already limited to "select" components.
```suggestion
    const baseDefaultText = this.intl.t('nrg.select.defaultText', {});
```